File: RestAPI101_Front/flutter_restapi101/lib/models/auth/authToken.dart
import 'package:flutter/material.dart';
import 'package:json_annotation/json_annotation.dart';

part 'authToken.g.dart';

@immutable
@JsonSerializable()
class AuthToken {
  @JsonKey(name: 'token')
  final String value;
  @JsonKey(required: true)
  final DateTime expires;

  AuthToken({required this.value, required this.expires});

  bool get valid => DateTime.now().toUtc().isBefore(expires);

  factory AuthToken.fromJson(Map<String, dynamic> json) => _$AuthTokenFromJson(json);
  Map<String, dynamic> toJson() => _$AuthTokenToJson(this);
}

File: RestAPI101.Backend/RestAPI101_Back/Models/AuthToken.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace RestAPI101_Back.Models
{
    public class AuthToken
    {
        [Required]
        public string Token { get; set; }
        [Required]
        public DateTime Expires { get; set; }

        public AuthToken(string token, DateTime expires)
        {
            Token = token;
            Expires = expires;
        }
    }
}
